### PR TITLE
Sorted imports using ISort

### LINKED ISSUE #12929: Sort imports
Sort imports in the package uniformly (see possible config in #12928) and introduce tests to check the files: `isort -rc -c -df` in CI.

### PR DESCRIPTION
**Pull request by jirikuncar**
closes #12929

### CODE REVIEW

**Comment by jirikuncar** [comment]
Sorting imports opened a pandora box of circular imports across the package. I will try to solve it if there is a general interested.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,25 @@
+root = true
```
Comment (asmeurer) [inline .editorconfig]: This file should not be here.

**Comment by asmeurer** [comment]
Circular imports are sadly a very common issue with SymPy. The current import order, especially in the code, is most likely very specifically there for a reason. 

A simpler change would be to sort the names being imported, not the import lines, as that doesn't affect anything.

**Comment by jirikuncar** [comment]
> A simpler change would be to sort the names being imported, not the import lines, as that doesn't affect anything.

I can have a look it this is possible with ISort. Otherwise, we can just mark files which cause problems.

File: .editorconfig (hunk 1 above)
Comment (jirikuncar) [inline .editorconfig]: It's from previous PR which provides the configuration for ISort.

**Pull request closed without merging**